create_db.py, accounting_cli.py: add configurable database path

cmoussa1 opened the pull request:
**Problem**: Both **create_db.py** and **accounting_cli.py** make assumptions about the location of the database file **FluxAccounting.db**. When **create_db.py** gets run, the database file gets placed in the same directory as the source code:

```
|-- LICENSE -> COPYING
|-- Makefile
|-- NOTICE -> NOTICE.LLNS
|-- NOTICE.LLNS
|-- README.md
|-- accounting
|   |-- FluxAccounting.db
|   |-- accounting_cli.py
|   |-- accounting_cli_functions.py
|   |-- create_db.py
```

This forces all of the subcommands to be executed in the same directory as well.

---

This PR adds a new optional command line argument `-p`, `--path` that allows the user to specify the database location when both creating the database and executing any of the subcommands. If no `-p` is passed, both scripts will default to the current directory (which was the behavior beforehand). 

An example of creating the database with the new argument:

```
$ ./create_db.py -p some/other/location/FluxAccounting.db
```

And running the subcommands:

```
$ flux account -p ../FluxAccounting.db view-user fluxuser
   id_assoc  creation_time    mod_time  deleted user_name  admin_level account parent_acct  shares  max_jobs  max_wall_pj
0         1     1593451150  1593451150        0  fluxuser            1    acct       pacct      10       100         1440
```

The subcommands will only open **FluxAccounting.db** in read-write mode instead of read-write-create. This should prevent empty database files from being created in case it cannot find the file. 

Fixes #20

cmoussa1 commented:
Okay, rebased after #19 and removed the **[WIP]** tag. I think this might be ready for some initial review

cmoussa1 commented:
Just pushed a fix to address some incorrect syntax with opening the .db file in URI mode; instead of `mode:rw`, it should be `mode=rw`.

SteVwonder commented on accounting/accounting_cli.py:
```diff
@@ -82,7 +87,19 @@ def main():
     args = parser.parse_args()
 
-    conn = sqlite3.connect("FluxAccounting.db")
+    # try to open database file; will exit with -1 if database file not found
+    if args.path:
+        try:
+            conn = sqlite3.connect("file:" + args.path + "?mode=rw", uri=True)
+        except sqlite3.OperationalError:
+            print("Unable to open database file")
+            sys.exit(-1)
+    else:
+        try:
+            conn = sqlite3.connect("file:FluxAccounting.db?mode=rw", uri=True)
+        except sqlite3.OperationalError:
+            print("Unable to open database file")
+            sys.exit(-1)
```
Nit, but you can eliminate some code duplication by leveraging a `path` variable. Something like:

```
        path = args.path if args.path else "FluxAccounting.db"
        try:
            conn = sqlite3.connect("file:" + path + "?mode=rw", uri=True)
        except sqlite3.OperationalError:
            print("Unable to open database file")
            sys.exit(-1)
```

SteVwonder commented on accounting/create_db.py:
```diff
@@ -47,7 +48,27 @@ def create_db(filepath):
+        except sqlite3.OperationalError:
+            print("Unable to create database file")
+    else:
+        create_db("FluxAccounting.db")
```
Does this create_db need to be wrapped in a try-catch as well?  Might want to leverage the suggestion above of leveraging a `path` variable.

SteVwonder commented on accounting/create_db.py:
```diff
@@ -47,7 +48,27 @@ def create_db(filepath):
+        try:
+            create_db(args.path)
+        except sqlite3.OperationalError:
+            print("Unable to create database file")
```
If the database fails to be created, I think this script should exit with a non-zero exit code.

cmoussa1 commented:
Thanks for taking the time to review this @SteVwonder! I believe I addressed your suggestions and pushed those up. Thanks for the suggestion about using a `path` variable; I hadn't seen that before.

cmoussa1 commented:
Thanks! :-)
